### ISSUE TITLE
ZFS datasets do not show up under 'Devices'

### ISSUE THREAD

**Issue by Redsandro** [linuxmint/nemo#1951]
```
 * Nemo version 3.8.6
 * Windowed nemo
 * Distribution - Mint 19
 * 64 bit
 ```

**Issue**

ZFS datasets do not show up under Nemo "Devices".

**Steps to reproduce**

Create some ZFS datasets with mountpoint `/media/<username>/<dataset>` and mount them.

**Expected behaviour**

Mounted datasets show up in Nemo under "Devices".

**Other information**

I have a Windows disk in my computer, which shows up under "Devices".
Any external drive shows up under "Devices".
When I mount a `sshfs` share, it shows up under "Devices".

ZFS datasets (mounts) should show up under "Devices" too.

PS - I know I can bookmark them so they show up under "My Computer". But they need to be in the category "Devices", just like other devices and mounts mounted under `/media/<username>/`.

---

![image](https://user-images.githubusercontent.com/1702193/45918567-29730180-be89-11e8-8987-fbbfb19baaff.png)

**Comment by mtwebster** [comment]
I'm unfamiliar with zfs, but with a bit of looking around I made a pool out of a file:
```
mtwebster:~> dd if=/dev/zero of=~/zfstest.img bs=1M count=1024
1024+0 records in
1024+0 records out
1073741824 bytes (1.1 GB, 1.0 GiB) copied, 0.498837 s, 2.2 GB/s

mtwebster:~> sudo zpool create pool-test ~/zfstest.img 
mtwebster:~> sudo zpool status
  pool: pool-test
 state: ONLINE
  scan: none requested
config:

	NAME                           STATE     READ WRITE CKSUM
	pool-test                      ONLINE       0     0     0
	  /home/mtwebster/zfstest.img  ONLINE       0     0     0

errors: No known data errors
```
And it showed up in nemo without extra prodding (though as read-only - I'll consider that a separate issue for now):
![image](https://user-images.githubusercontent.com/262776/45922461-96db5e00-be9a-11e8-8afe-ca9fec2bb103.png)

However, you said 'dataset' and not pool - do you want to be able to access individual members of the pool?  Isn't this counter to the whole reason of this (it seems to me like a software raid-type system)?  Isn't the pool itself considered a dataset?

Can you please explain further?  I don't have spare physical drives available, so if possible provide me a case/explanation that can be worked on in the context of a file-based setup.

**Comment by Redsandro** [comment]
@mtwebster that's interesting. I didn't even know you could use a `pool` without creating `datasets`. I've never seen it used like that. I'm not even sure it's a ZFS file system at this stage, because for datasets you use the `zfs` command.

---

### Off-topic

I am guessing that a `pool` is _like_ a disk, and a `dataset` is _like_ a partition. You can have one `dataset` for every user on a server, but ZFS data deduplication is pool-wide. 

In other words, if we shared a machine, I can have `pool/redsandro` mounted at `/media/redsandro/zfs`, you can have `pool/mtwebster` mounted at `/media/mtwebster/zfs`, and if we both have `linuxmint-19-cinnamon-64bit-v2.iso` downloaded, it would (eventually) be stored on the pool only once.

---

Anyway, the fact that your pool shows up is interesting too, as for me, the pool does not show up either. However, my pool is mounted in `/`, not in `/media/redsandro`. So I am not surprised, because as far as I know, "devices" need to be mounted in `/media` in order to show up in Nemo or Nautilus.

Could you please test this out for me by creating a dataset? 

```
sudo zfs create -o mountpoint=/media/<username>/dataset-test pool-test/dataset-test
```

If they too show up, you're on to something.

**Comment by Redsandro** [comment]
__Note to self:__ Normal shares show up in the manager when `x-gvfs-show` is set as a mount option or comment. Can we specify x- custom options with zfs?

@mtwebster You didn't have time yet to verify if datasets are also automatically showing up in the file manager. Assuming they do, I'm hypothesizing that you might be using the Ubuntu default implementation of ZFS 7.5. I'm using the official/vanilla (pre-)releaase of ZFS 8.

This might indicate that Ubuntu has implemented some kind of hook that the vanilla release doesn't have. I'm trying to find out if so, and _what_ that hook is, so that I may recreate similar functionality with the vanilla ZFS.

It would also mean that this issue is unrelated to Nemo.

**Comment by mtwebster** [comment]
Sorry, I'm easily distracted - 

Weird stuff here.. `same computer from my original test.  `sudo zpool status` says no pools available.  I haven't done anything related to this since my report above. 

Repeating the procedure also failed to add another drive.  Though now the status command shows the new one I made.  I performed the command you requested and it succeeded, but nothing in nemo.

I finally realized that drive was a one of two partitions I had made to test this out initially.  For some reason I changed gears and used files instead.  I feel like an idiot now, sending you on a wild goose chase.  Sorry about that.

Is there any value in zfs-fuse?  I only mention it because there's a gvfs backend for fuse.  Nemo(gio) uses gvfs for mounts, etc..

I quickly tried `zpool create pool /dev/sda8`, an empty 1gb drive.  It succeeded:
```
mtwebster:~> sudo zpool list
NAME   SIZE  ALLOC   FREE    CAP  DEDUP  HEALTH  ALTROOT
pool  1016M   444K  1016M     0%  1.00x  ONLINE  -
```
but nothing in nemo (though the drive itself shows up, the same as it did prior to creating this pool - with the addition of a trash folder now)

**Comment by xenopeek** [comment]
I think the issue here was zpool creates the mountpoint for the pool in /poolname by default and Nemo doesn't see that as removable device location. The solution, if you want to have it show up under Devices, is to set your mountpoint accordingly.

I did the following to test:

```
sudo mkdir /media/<user>/<mountpoint>
sudo zpool create -m /media/<user>/<mountpoint> <poolname> /dev/sd<n>
sudo chown -R <user>: /media/<user>/<mountpoint>
```

I called my mountpoint "zfstest" and it shows up as expected and is writable (from the last command):
![image](https://user-images.githubusercontent.com/21159972/85939937-a44c3480-b919-11ea-99c0-027dd5590856.png)
